### PR TITLE
wip snappoints #91

### PR DESCRIPTION
**Pull request by jogibear9988**
# Types of Changes

## Prerequisites

Please make sure you can check the following two boxes:

- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project

## Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
- [x] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [ ] All new and existing tests passed

## Description

Add CSS Snappoints Properties

### CODE REVIEW

**Comment by jogibear9988** [comment]
WIP, doesn't even compile.

I'm not sure how to do the conversation for the ScrollSnaptype, cause there are two enums wich combine the functionality...

Can we Generate the converter by Combinding "SeperatorConverter" and Enum Converters?
Do we need the new Constant<T,U> Type? Most is only a Test and for discussion...

I need to look, maybe I find another CssProperty wich consists of two enums, and I see how it's done there...

**Comment by jogibear9988** [comment]
@FlorianRappl 
maybe you can look and give a few hint's how you like it to be solved...

File: src/AngleSharp.Css/Converters/SeparatedEnumsConverter.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        public SeparatedEnumsConverter(IValueConverter[] converters, Char seperator)
+        {
+            _converters = converters;
+            _seperator = seperator;
+        }
+
+        public ICssValue Convert(StringSource source)
+        {
+            var value = _converters[0].Convert(source);
+            new Constant<T,U>()
```
Comment (FlorianRappl) [inline src/AngleSharp.Css/Converters/SeparatedEnumsConverter.cs:23]: Is this here by mistake?

File: src/AngleSharp.Css/Values/Primitives/Constant2.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
+namespace AngleSharp.Css.Values
```
Comment (FlorianRappl) [inline src/AngleSharp.Css/Values/Primitives/Constant2.cs:1]: Personally, I'd prefer having both definitions in the same file.

Alternatively, we could rename the first file (`Constant.ts`) to `Constant1.cs`.

File: src/AngleSharp.Css/Values/Primitives/Constant2.cs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+{
+    using System;
+
+    /// <summary>
+    /// Represents a selected CSS enum value.
+    /// </summary>
+    struct Constant<T,U> : ICssPrimitiveValue, IEquatable<Constant<T,U>>
```
Comment (FlorianRappl) [inline src/AngleSharp.Css/Values/Primitives/Constant2.cs:8]: Is this really used? I only see one line with 2 types and that is not really used (see question).

**Pull request closed without merging**